sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/ProactiveOpenConnectionsProcessor.java

```diff
@@ -30,15 +30,18 @@
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
 import java.util.stream.Collectors;
 
 public final class ProactiveOpenConnectionsProcessor implements Closeable {
 
     private static final Logger logger = LoggerFactory.getLogger(ProactiveOpenConnectionsProcessor.class);
     private Sinks.Many<OpenConnectionTask> openConnectionsTaskSink;
```

Comment:
In `recordOpenConnectionsAndInitCachesStarted` we dispose the sink and instantiate it again `getBackgroundOpenConnectionsPublisher` do we need to instantiate the sink in the constructor?

Reply:
make sense. I removed from the constructor
but we will need to start the background tasks when warm up flow is not used at all, so I added back the init method